frontend/src/AppBuilder/_stores/slices/eventsSlice.js

```diff
@@ -504,6 +506,38 @@ export const createEventsSlice = (set, get) => ({
             }
             return Promise.resolve();
           }
+          case 'log-info': {
+            get().eventsSlice.logError(
+              'Custom Log',
+              'Custom-log',
+              '',
+              eventObj,
+              {
+                eventId: event.eventId,
+              },
+              'success'
+            );
+            break;
+          }
+          case 'log': {
+            get().eventsSlice.logError(
+              'Custom Log',
+              'Custom-log',
+              '',
+              eventObj,
+              {
+                eventId: event.eventId,
+              },
+              'success'
+            );
+            break;
+          }
+          case 'log-error': {
+            get().eventsSlice.logError('Custom Log', 'Custom-log', '', eventObj, {
+              eventId: event.eventId,
+            });
```

Comment:
The code is repeated for the three cases mostly. Can we reuse using a function or variable?